src/Http/Routing/src/Builder/RouteHandlerFilterExtensions.cs

```diff
@@ -29,9 +30,16 @@ public static RouteHandlerBuilder AddFilter(this RouteHandlerBuilder builder, IR
     /// <typeparam name="TFilterType">The type of the <see cref="IRouteHandlerFilter"/> to register.</typeparam>
     /// <param name="builder">The <see cref="RouteHandlerBuilder"/>.</param>
     /// <returns>A <see cref="RouteHandlerBuilder"/> that can be used to further customize the route handler.</returns>
-    public static RouteHandlerBuilder AddFilter<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicConstructors)] TFilterType>(this RouteHandlerBuilder builder) where TFilterType : IRouteHandlerFilter, new()
+    public static RouteHandlerBuilder AddFilter<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicConstructors)] TFilterType>(this RouteHandlerBuilder builder) where TFilterType : IRouteHandlerFilter
     {
-        builder.RouteHandlerFilters.Add(new TFilterType());
+        // We can provide the `MethodInfo` as an argument to the factory here so that `IRouteHandlerFilter`
```

Comment:
Is methodInfo supposed to be used somewhere based on this comment?

Reply:
This was meant to be a to-do from an earlier discussion.

But as we discussed in the chat, we're just going to allow passing `RouteHandlerContext` here.